src/core_modules/capture-core/components/Pages/common/WidgetEventEditWrapper/WidgetEventEditWrapper.js

```diff
@@ -26,9 +36,21 @@ export const WidgetEventEditWrapper = ({ pageStatus, ...passOnProps }: WidgetPro
         );
     }
 
+    if (isLoading || !formFoundation || !stage || isError) {
+        return (
+            <div>
+                {i18n.t('Loading')}
+            </div>
+        );
+    }
```

Comment:
We haven't really done loading states to the full extent that it could, but do we have a consensus on what we should do in this case? Many other places we just display "X can't be loaded" or something resembling an error message, but that does not feel quite right either.